pkg/mcoproducer/mcoproducer.go

```diff
@@ -0,0 +1,146 @@
+    podman pull --authfile /var/lib/kubelet/config.json $IMAGE:$TAG
+    if [ $? -eq 0 ]; then
+        echo "Image $IMAGE:$TAG has been successfully pulled, rebooting.."
+        reboot
+    else
+        echo "Failed to pull image $IMAGE:$TAG"
+    fi
+fi
+`
+
+const mcoTemplate = `apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: $MCO_MACHINE_CONFIG_POOL_REF
+  name: $MCO_NAME
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    systemd:
+      units:
+        - contents: |
+            [Unit]
+            Description=Replace in-tree kernel module with oot kernel module
+            Before=network-pre.target
+            Wants=network-pre.target
+            DefaultDependencies=no
+            [Service]
+            User=root
+            Type=oneshot
+            TimeoutSec=10
+            ExecStartPre=ls /usr/local/bin
+            ExecStart=/usr/local/bin/replace-kernel-module.sh
+            PrivateTmp=yes
+            RemainAfterExit=no
+            TimeoutSec=60
+            [Install]
+            WantedBy=multi-user.target
+          enabled: true
+          name: "replace-kernel-module.service"
+        - contents: |
+            [Unit]
+            Description=Pull oot kernel module image
+            After=network-online.target
+            Wants=network-online.target
+            DefaultDependencies=no
+            [Service]
+            User=root
+            Type=oneshot
+            ExecStart=/usr/local/bin/pull-kernel-module-image.sh
+            PrivateTmp=yes
+            RemainAfterExit=no
+            TimeoutSec=900
+            [Install]
+            WantedBy=multi-user.target
+          enabled: true
+          name: "pull-kernel-module-image.service"
+        - enabled: false
+          mask: true
+          name: crio-wipe.service
+    storage:
+      files:
+        - path: "/usr/local/bin/replace-kernel-module.sh"
+          mode: 511
+          overwrite: true
+          user:
+            name: "root"
+          contents:
+            source: "data:text/plain;base64,$REPLACE_IN_TREE_DRIVER_CONTENTS"
+        - path: "/usr/local/bin/pull-kernel-module-image.sh"
+          mode: 493
+          overwrite: true
+          user:
+            name: "root"
+          contents:
+            source: "data:text/plain;base64,$PULL_OOT_KERNEL_MODULE_CONTENTS"
+`
+
+func ProduceMCO(machineConfigName, machineConfigPoolRef, kernelModuleImage, kernelModuleName string) (string, error) {
+	tag, err := name.NewTag(kernelModuleImage)
+	if err != nil {
+		return "", fmt.Errorf("invalid kernelModuleImage %s, input should be repo:tag format: %v", kernelModuleImage, err)
+	}
+
+	scriptsVars := []string{
+		"IMAGE=" + tag.Repository.Name(),
+		"TAG=" + tag.Identifier(),
+		"KERNEL_MODULE=" + kernelModuleName,
+	}
+
+	generatedScripts, err := utils.ReplaceInTemplates(scriptsVars, replaceKernelModuleScriptTemplate, pullKernelModuleScriptTemplate)
```

Comment:
What do we have such function in our code? `ReplaceInTemplates`.
Why don't we use [golang tempaltes](https://pkg.go.dev/text/template) for that?

Reply:
we already have utils package that deals with replacing the env variable in KernelMapping and Dockerfile. So instead of using golang templates, i am using already existing package

Reply:
I think we should go with Go templates as they are part of the standard library and can be parsed only once.